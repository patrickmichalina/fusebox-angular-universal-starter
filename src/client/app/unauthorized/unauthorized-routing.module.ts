import { UnauthorizedComponent } from './unauthorized.component'
import { NgModule } from '@angular/core'
import { RouterModule } from '@angular/router'
import { MetaGuard } from '@ngx-meta/core'

@NgModule({
  imports: [
    RouterModule.forChild([
      {
        path: '',
        component: UnauthorizedComponent,
        canActivate: [MetaGuard],
        data: {
          meta: {
            title: 'i18n.unauthorized.title',
            description: 'i18n.unauthorized.description'
          }
        }
      }
    ])
  ],
  exports: [RouterModule]
})
export class UnauthorizedRoutingModule { }
