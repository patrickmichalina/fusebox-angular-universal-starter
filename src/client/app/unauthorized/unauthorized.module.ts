import { UnauthorizedRoutingModule } from './unauthorized-routing.module'
import { UnauthorizedComponent } from './unauthorized.component'
import { NgModule } from '@angular/core'
import { SharedModule } from '../shared/shared.module'

@NgModule({
  imports: [UnauthorizedRoutingModule, SharedModule],
  declarations: [UnauthorizedComponent],
  exports: [UnauthorizedComponent]
})
export class UnauthorizedModule { }
