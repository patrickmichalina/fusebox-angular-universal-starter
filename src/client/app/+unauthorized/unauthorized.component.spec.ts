import { REQUEST } from '@nguniversal/express-engine/tokens'
import { AngularFireAuthModule } from 'angularfire2/auth'
import { UnauthorizedComponent } from './unauthorized.component'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { Component } from '@angular/core'
import { UnauthorizedModule } from './unauthorized.module'
import { AngularFireModule } from 'angularfire2'

describe(UnauthorizedComponent.name, () => {
  let fixture: ComponentFixture<UnauthorizedComponent>

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [UnauthorizedModule, AngularFireAuthModule, AngularFireModule.initializeApp({
        'apiKey': '1',
        'authDomain': 'app.firebaseapp.com',
        'databaseURL': 'https://app.firebaseio.com',
        'projectId': 'firebase-app',
        'messagingSenderId': '1'
      })],
      declarations: [TestComponent],
      providers: [
        { provide: REQUEST, useValue: {} }
      ]
    }).compileComponents()
  }))

  beforeEach(async(() => {
    fixture = TestBed.createComponent(UnauthorizedComponent)
  }))

  afterEach(async(() => {
    TestBed.resetTestingModule()
  }))

  it('should compile', async(() => {
    expect(fixture.nativeElement).toBeDefined()
    expect(fixture).toMatchSnapshot()
  }))
})

@Component({
  selector: 'test-component',
  template: '<pm-logout></pm-logout>'
})
class TestComponent { }
