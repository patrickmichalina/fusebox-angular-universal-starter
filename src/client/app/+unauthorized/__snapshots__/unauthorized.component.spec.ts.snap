// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnauthorizedComponent should compile 1`] = `
<pm-unauthorized>
  <div
    fxlayout="column"
    fxlayoutalign="center center"
  >
    
    
    <h1>
      Unauthorized
    </h1>
    
    
    <pm-login-card>
      
      
  
      
    </pm-login-card>
    
  
  </div>
</pm-unauthorized>
`;
